### PR TITLE
Add fspacectl, new in FreeBSD 14

### PR DESCRIPTION
**Pull request by asomers**

### CODE REVIEW

**Comment by JohnTitor** [comment]
Finally we should prepare FreeBSD 14 CI, let me do that before merging.

**Comment by JohnTitor** [comment]
Needs a rebase.

**Comment by asomers** [comment]
Rebased, @JohnTitor .

File: src/unix/bsd/freebsdlike/freebsd/mod.rs [hunk 1]
```diff
@@ -1098,6 +1103,28 @@ cfg_if! {
...
+                self.r_len.hash(state);
+            }
+        }
+        impl Eq for spacectl_range {}
+        impl PartialEq for spacectl_range {
+            fn eq(&self, other: &Self) -> bool {
+                self.r_offset == other.r_offset
+                    && self.r_len == other.r_len
+            }
+        }
```
Comment (Amanieu) [inline src/unix/bsd/freebsdlike/freebsd/mod.rs:1126]: Can't you just `#[derive]` these?

**Comment by asomers** [comment]
I rebased the changes, derived the extra traits, and made spacectl_range's fields public.

**Pull request merged**